common/djangoapps/student/views/management.py

```diff
@@ -701,6 +705,39 @@ def create_account_with_params(request, params):
                 request.social_strategy.clean_partial_pipeline(social_access_token)
                 raise ValidationError({'access_token': [error_message]})
 
+        # If the user is registering via 3rd party auth, track which provider they use
+        if is_third_party_auth_enabled and pipeline.running(request):
+            running_pipeline = pipeline.get(request)
+            third_party_provider = provider.Registry.get_from_pipeline(running_pipeline)
```

Comment:
Does this code need to be in the transaction?  Seems like it should be all selects based on data available before any DB changes are made

Reply:
I wasn't sure if any implementations modified tables as part of the pipeline so I played it safe. I'm not familiar enough with the internals to say for sure.